upgrade/scripts/k8s/upgrade_control_plane.sh

```diff
@@ -27,6 +27,7 @@ echo ""
 kubectl get configmap kubeadm-config -n kube-system -o yaml > /tmp/kubeadm-config.yaml
 cp /tmp/kubeadm-config.yaml /tmp/kubeadm-config.yaml.back
 sed -i 's/imageRepository: k8s.gcr.io/imageRepository: artifactory.algol60.net\/csm-docker\/stable\/k8s.gcr.io/' /tmp/kubeadm-config.yaml
+sed -i 's/    dns: {}/    dns:\n      type: CoreDNS\n      imageRepository: artifactory.algol60.net\/csm-docker\/stable\/k8s.gcr.io\/coredns/' /tmp/kubeadm-config.yaml
```

Comment:
Something like this would be a little more flexible
```suggestion
sed -i -E 's/([[:space:]]+)dns: {}/\1\n\1  type: CoreDNS\n\1  imageRepository: artifactory.algol60.net\/csm-docker\/stable\/k8s.gcr.io\/coredns/' /tmp/kubeadm-config.yaml
```

Reply:
good suggestion, changed and tested on drax (with a slight tweak)